src/parser/core/modules/EntityStatuses.ts

```diff
@@ -162,12 +162,18 @@ export class EntityStatuses extends Module {
 		if (statusEvent.end == null) {
 			this.debug('Unfinished status event detected.  Applying ability duration after last refresh event.')
 			const statusInfo = this.data.getStatus(statusEvent.ability.guid)
+			let end: number
 			if (statusInfo?.duration) {
-				statusEvent.end = Math.min(this.parser.fight.end_time, statusEvent.lastRefreshed + statusInfo.duration * 1000)
-				this.debug(`Updating status event for status ${statusInfo.name}.  Adding ${statusInfo.duration} seconds, effective end time set to ${this.parser.formatTimestamp(statusEvent.end, 1)}`)
+				end = Math.min(this.parser.currentTimestamp, statusEvent.lastRefreshed + statusInfo.duration * 1000)
+				this.debug(`Updating status event for status ${statusInfo.name}.  Adding ${statusInfo.duration} seconds, effective end time set to ${this.parser.formatTimestamp(end, 1)}`)
 			} else {
 				this.debug(`No matching status duration information found for status ${statusEvent.ability.guid}, setting to end of fight so invuln detection can clip the end to when the target went untargetable`)
-				statusEvent.end = this.parser.pull.duration + this.parser.eventTimeOffset
+				end = this.parser.currentTimestamp
+			}
+
+			statusEvent = {
+				...statusEvent,
+				end,
```

Comment:
While this is primarily called at the end of the fight, tightening usage anyway. I'm limiting to the current timestamp, rather than the end of the fight, which could be in the future.

I'm also shallow cloning the `end` in so it doesn't pollute potential other consumers with this data.